### PR TITLE
Add optional feature to use cache-key-url for parent selection

### PR DESCRIPTION
**Pull request by JBevillC**
This feature allows more than one parent cache to be selected for ranges requested from the same URL.

### CODE REVIEW

**Comment by jrushford** [comment]
@JBevillC you need to run clang-format.

**Comment by JBevillC** [comment]
Updated with clang-format

**Comment by jrushford** [comment]
[approve ci]

**Comment by JBevillC** [comment]
Squashed and re-pushed.   Successful compile locally with devtoolset-7 on Centos7

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Comment by jrushford** [comment]
[approve ci]

File: plugins/experimental/cache_range_requests/cache_range_requests.cc [hunk 1]
```diff
@@ -35,18 +35,76 @@
...
+    ERROR_LOG("Can't allocate pluginconfig");
+    return nullptr;
+  }
+
+  // Plugin uses default ATS selection (hash of URL path)
+  pc->ps_mode = PS_DEFAULT;
+
+  // Walk through param list.
+  for (int c = 0; c < argc; c++) {
+    if (strcmp("ps_mode:cache_key_url", argv[c]) == 0) {
```
Comment (alficles) [inline plugins/experimental/cache_range_requests/cache_range_requests.cc]: This appears to be parsing the "to" and "from" mappings as part of the parameters, which is probably harmless, but probably also incorrect.

Comment (jrushford) [reply plugins/experimental/cache_range_requests/cache_range_requests.cc]: You should probably initialize c = 2 @JBevillC

Comment (JBevillC) [reply plugins/experimental/cache_range_requests/cache_range_requests.cc]: Thanks for finding this small bug Chris.  Squashed and re-pushed.

**Comment by JBevillC** [comment]
I updated the branch to skip over arguments that shouldn't be parsed by the create_pluginconfig() function.

**Comment by jrushford** [comment]
[approve ci]

**Comment by zwoop** [comment]
Cherry-picked to 8.0.x

**Pull request merged**